### ISSUE TITLE
Is --publish option really needed for documentation upload?

### ISSUE THREAD

**Issue by amigalemming** [haskell/cabal#4777]
I try:
~~~~
$ cabal upload --documentation dist/llvm-extra-0.7.3-docs.tar.gz
Hackage password: 
Uploading documentation dist/llvm-extra-0.7.3-docs.tar.gz...
Ok
~~~~
Cabal-install says "Ok", but documentation does not appear on Hackage.
It does not mention something about "documentation candidates" or so.

I try:
~~~~
cabal upload --documentation --publish dist/llvm-extra-0.7.3-docs.tar.gz
Hackage password: 
Uploading documentation dist/llvm-extra-0.7.3-docs.tar.gz...
Ok
~~~~
and now the documentation appears online.

Is it intended that I must use the `--publish` flag? Are there "documentation candidates" and are they needed? I can always replace documentation with remakes of the documentation, so the situation is less problematic than uploading new (immutable) package versions.

**Comment by phadej** [comment]
ping @hvr 

to my understanding you can upload docs to candidate packages too, and I'm +1 for having consistent UI, i.e. having `cabal upload --documentation --candidate` isn't uniform.

**Comment by amigalemming** [comment]
On Wed, 20 Sep 2017, Oleg Grenrus wrote:

> ping @hvr
> 
> to my understanding you can upload docs to candidate packages too, and I'm +1 for having consistent UI, i.e.
> having cabal upload --documentation --candidate isn't uniform.

Then I suggest that cabal-install tells me that the documentation is not 
yet published (like when uploading a package) and how to publish it.

**Comment by 23Skidoo** [comment]
> Then I suggest that cabal-install tells me that the documentation is not yet published (like when uploading a package) and how to publish it.

+1 from me.

**Comment by 23Skidoo** [comment]
This was fixed by #4779. Please reopen if you think that there's more to be done here.
